### PR TITLE
Issue #432: Remove exported filtering functions

### LINKED ISSUE #432: Remove `filter_obs()` (if possible)
Like #431 I think we want to get rid of we can.

### PR DESCRIPTION
**Pull request by athowes**
## Description

This PR closes #432.

It moves one filtering function into the two places it's used. And deletes the other which is used no where.

We could have follow up issue on adding back in some better simulation code about this, but we could also do that later.

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: vignettes/approx-inference.Rmd [hunk 1]
```diff
@@ -110,6 +110,15 @@ sdlog <- 0.5
 obs_time <- 25
 sample_size <- 200
 
+filter_obs_by_obs_time <- function(linelist, obs_time) {
```
Comment (seabbs) [inline vignettes/approx-inference.Rmd:113]: I really don't like putting function definitions into the vignettes like this. I think all we need is the filter line so can you just use that as needed vs this approach

Comment (athowes) [reply vignettes/approx-inference.Rmd:113]: Sure I agree that is probably better. This is quicker to do now and could be quicker when we do the refactor to a future function

**Comment by athowes** [comment]
@seabbs lmk if you want me to take back over on getting this passing tests

**Review by seabbs — APPROVED** [review]
I think this looks okay now

**Pull request merged**